webapp/src/ts/services/task-notifications.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@angular/core';
+import * as moment from 'moment';
+import { orderBy } from 'lodash-es';
+
+import { RulesEngineService } from '@mm-services/rules-engine.service';
+import { TranslateService } from '@mm-services/translate.service';
+import { DBSyncService } from '@mm-services/db-sync.service';
+
+/* 
+** avoid overloading app with too many notifications especially at once
+** 24 for android >= 10
+*/
+const MAX_NOTIFICATIONS = 24;
+const TODAY_TIMESTAMP = 'cht-today-timestamp';
+const LATEST_NOTIFICATION_TIMESTAMP = 'cht-latest-notification-timestamp';
+
+export interface Notification {
+  _id: string,
+  authoredOn: number,
+  state: string,
+  title: string,
+  contentText: string,
+  dueDate: string,
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TasksNotificationService {
+
+  constructor(
+    private readonly rulesEngineService: RulesEngineService,
+    private readonly translateService: TranslateService,
+    private readonly dbSyncService: DBSyncService
+  ) { }
+
+  private async fetchNotifications(): Promise<Notification[]> {
+    try {
+      const today = moment().format('YYYY-MM-DD');
+      let latestNotificationTimestamp = this.getLatestNotificationTimestamp();
+      const isEnabled = await this.rulesEngineService.isEnabled();
+      const taskDocs = isEnabled ? await this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
+      
+      let notifications = taskDocs
+        .filter(task => {
+          return task.state === 'Ready' && task.emission.dueDate === today && 
+            task.authoredOn > latestNotificationTimestamp;
```

Comment:
I'm not sure what purpose this serves. `authoredOn` is when the task gets first created. This can be up to 60 days before the task is ready. How is the date when the task doc was **created** relevant for showing notifications?